scripts/bert/create_pretraining_data.py

```diff
@@ -0,0 +1,464 @@
+
+parser.add_argument(
+    '--vocab_file',
+    type=str,
+    default=None,
+    help='The vocabulary file that the BERT model was trained on.')
+
+parser.add_argument(
+    '--do_lower_case',
+    action='store_true',
+    help='Whether to lower case the input text. Should be True for uncased'
+    'models and False for cased models.')
+
+parser.add_argument(
+    '--max_seq_length', type=int, default=128, help='Maximum sequence length.')
+
+parser.add_argument(
+    '--max_predictions_per_seq',
+    type=int,
+    default=20,
+    help='Maximum number of masked LM predictions per sequence.')
+
+parser.add_argument(
+    '--random_seed',
+    type=int,
+    default=12345,
+    help='Random seed for data generation.')
+
+parser.add_argument(
+    '--dupe_factor',
+    type=int,
+    default=10,
+    help='Number of times to duplicate the input data (with different masks).')
+
+parser.add_argument(
+    '--masked_lm_prob',
+    type=float,
+    default=0.15,
+    help='Masked LM probability.')
+
+parser.add_argument(
+    '--short_seq_prob',
+    type=float,
+    default=0.1,
+    help='Probability of creating sequences which are shorter than the '
+    'maximum length.')
+
+args = parser.parse_args()
+
+logging.info(args)
+
+
+class TrainingInstance:
+    """A single training instance (sentence pair)."""
+
+    def __init__(self, tokens, segment_ids, masked_lm_positions,
+                 masked_lm_labels, is_random_next):
+        self.tokens = tokens
+        self.segment_ids = segment_ids
+        self.is_random_next = is_random_next
+        self.masked_lm_positions = masked_lm_positions
+        self.masked_lm_labels = masked_lm_labels
+
+    def __str__(self):
+        s = ''
+        s += 'tokens: %s\n' % (' '.join(
+            [tokenization.printable_text(x) for x in self.tokens]))
+        s += 'segment_ids: %s\n' % (' '.join(
+            [str(x) for x in self.segment_ids]))
+        s += 'is_random_next: %s\n' % self.is_random_next
+        s += 'masked_lm_positions: %s\n' % (' '.join(
+            [str(x) for x in self.masked_lm_positions]))
+        s += 'masked_lm_labels: %s\n' % (' '.join(
+            [tokenization.printable_text(x) for x in self.masked_lm_labels]))
+        s += '\n'
+        return s
+
+    def __repr__(self):
+        return self.__str__()
+
+
+def write_instance_to_example_files(instances, tokenizer, max_seq_length,
+                                    max_predictions_per_seq, output_files):
+    """Create RecordIO files from `TrainingInstance`s."""
+    writers = []
+    for output_file in output_files:
+        writers.append(mx.recordio.MXRecordIO(output_file, 'w'))
+
+    writer_index = 0
+
+    total_written = 0
+    for (inst_index, instance) in enumerate(instances):
+        input_ids = tokenizer.convert_tokens_to_ids(instance.tokens)
+        input_mask = [1] * len(input_ids)
+        segment_ids = list(instance.segment_ids)
+        assert len(input_ids) <= max_seq_length
+
```

Comment:
For padded data, for consistency let's put 
```
pad = tokenizer.convert_tokens_to_ids('[PAD]')
...
            input_ids.append(pad)
            input_mask.append(pad)
            segment_ids.append(pad)

```

Reply:
Can we be sure padding of  input_mask, segment_ids will be same?

input_mask, segment_ids will have only '0', '1'

This will be correct.

```
pad = tokenizer.convert_tokens_to_ids('[PAD]')
...
            input_ids.append(pad)
            input_mask.append(0)
            segment_ids.append(0)
```

Reply:
Oh you're right. mask and seg_id should remain 0